QLoggerWriter.h

```diff
@@ -0,0 +1,179 @@
+    * @return The maximum size
+    */
+   int getMaxFileSize() const { return mMaxFileSize; }
+
+   /**
+    * @brief setMaxFileSize Sets the max file size for this destination.
+    * @param maxSize The new file size
+    */
+   void setMaxFileSize(int maxSize) { mMaxFileSize = maxSize; }
+
+   /**
+    * @brief getMessageOptions Gets the current message options.
+    * @return The current options
+    */
+   LogMessageDisplays getMessageOptions() const { return mMessageOptions; }
+
+   /**
+    * @brief setMessageOptions Specifies what elements are displayed in one line of log message.
+    * @param messageOptions The options
+    */
+   void setMessageOptions(LogMessageDisplays messageOptions) { mMessageOptions = messageOptions; }
+
+   /**
+    * @brief enqueue Enqueues a message to be written in the destiantion.
+    * @param date The date and time of the log message.
+    * @param threadId The thread where the message comes from.
+    * @param module The module that writes the message.
+    * @param level The log level of the message.
+    * @param function The function that prints the log.
+    * @param fileName The file name that prints the log.
+    * @param line The line of the file name that prints the log.
+    * @param message The message to log.
+    */
+   void enqueue(const QDateTime &date, const QString &threadId, const QString &module, LogLevel level, const QString &function,
+                const QString &fileName, int line, const QString &message);
+
+   /**
+    * @brief Stops the log writer
+    * @param stop True to be stop, otherwise false
+    */
+   void stop(bool stop) { mIsStop = stop; }
+
+   /**
+    * @brief Returns if the log writer is stop from writing.
+    * @return True if is stop, otherwise false
+    */
+   bool isStop() const { return mIsStop; }
+
+   void run() override;
+
+   void closeDestination();
+
+private:
+   bool mQuit = false;
+   bool mIsStop = false;
+   QWaitCondition mQueueNotEmpty;
+   QString mFileDestinationFolder;
+   QString mFileDestination;
+   LogFileDisplay mFileSuffixIfFull;
+   LogMode mMode;
+   LogLevel mLevel;
+   int mMaxFileSize = 1024 * 1024;  //! @note 1Mio
+   LogMessageDisplays mMessageOptions;
+   QVector<QPair<QString, QString>> messages;
+   QMutex mutex;
+
+   /**
+    * @brief renameFileIfFull Truncates the log file in two. Keeps the filename for the new one and renames the old one
+    * with the timestamp or with a file number.
+    *
+    * @return Returns the file name for the old logs.
+    */
+   QString renameFileIfFull();
+
+   /**
+    * @brief generateDuplicateFilename
+    *
+    * @param fileDestination The file path and name without the extension.
+    * @param fileExtension The file extension
+    * @param p_number The file suffix number.
+    * @return The complete path of the duplicated file name.
+    */
+   static QString generateDuplicateFilename(const QString& fileDestination, const QString& fileExtension, int p_number=1);
```

Comment:
Could we use fileSuffixNumber instead of p_number? :)

Reply:
Sure ! Sorry, "p_" prefixes are usually my way of coding